src/services/completions.ts

```diff
@@ -4299,6 +4298,34 @@ namespace ts.Completions {
             }
             return undefined;
         }
+
+    }
+
+    function getSingleLineReplacementSpanForImportCompletionNode(node: ImportDeclaration | ImportEqualsDeclaration | ImportSpecifier | Token<SyntaxKind.ImportKeyword> | undefined) {
+        if (!node) return undefined;
+        const top = (findAncestor(node, or(isImportDeclaration, isImportEqualsDeclaration)) || node) as ImportDeclaration | ImportEqualsDeclaration | Token<SyntaxKind.ImportKeyword>;
+        const sourceFile = top.getSourceFile();
+        if (rangeIsOnSingleLine(top, sourceFile)) {
+            return createTextSpanFromNode(top, sourceFile);
+        }
+        // Import keyword alone was necessarily on one line
+        Debug.assert(top.kind !== SyntaxKind.ImportKeyword);
+        const withoutModuleSpecifier: TextRange = {
+            pos: top.getFirstToken()!.getStart(),
+            end: top.kind === SyntaxKind.ImportDeclaration ? top.moduleSpecifier.pos : top.moduleReference.pos,
+        };
+        // The module specifier/reference was previously found to be missing, empty, or
+        // not a string literal - in this last case, it's likely that statement on a following
+        // line was parsed as the module specifier of a partially-typed import, e.g.
+        //   import Foo|
+        //   interface Blah {}
+        // This appears to be a multiline-import, and editors can't replace multiple lines.

```

Comment:
> editors can't replace multiple lines

Is this detail actually relevant to the change? It sounds like even if editors could do that, you would still want to assume that the next line is a separate statement and recover appropriately, right?

Reply:
Yeah, fair; but it explains why the failure mode is “no completions” rather than “overwriting later declarations.”